spi: pass correct buffer to spi driver given NULL to spi_transceive

wentongwu opened the pull request:
When pass NULL to spi_transceive with user space enabled, stack buffer
is still passed to spi driver and it will cause kinds of problems like
MPU fault, so change it to pass relevant NULL pointers in the actual
transceive call.

Also fixes a spell typo.

Fixes: #20811.

tbursztyka commented on drivers/spi/spi_handlers.c:
```diff
@@ -11,15 +11,14 @@
-			   struct spi_buf *buf_copy,
-			   int writable)
+static struct spi_buf_set *copy_and_check(struct spi_buf_set *bufs,
+			   struct spi_buf *buf_copy, int writable)
```
indent the parameters to last '(' + 1 space (and keep writable parameter on its own line as well)

wentongwu replied:
@tbursztyka updated.

tbursztyka replied:
thanks

tbursztyka reviewed (approved):
just a tiny style issue. rest is fine so approving

wentongwu committed:
spi: pass correct buffer to spi driver given NULL to spi_transceive

When pass NULL to spi_transceive with user space enabled, stack buffer
is still passed to spi driver and it will cause kinds of problems like
MPU fault, so change it to pass relevant NULL pointers in the actual
transceive call.

Fixes: #20811.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>

wentongwu committed:
spi: fix spell typo in spi handler

fix spell typo in spi handler.

Signed-off-by: Wentong Wu <wentong.wu@intel.com>
